### PR TITLE
chore: update mix.exs dependencies

### LINKED ISSUE #111: Deprecation warnings
The following warnings are observed on 
Elixir 1.8.1 (compiled with Erlang/OTP 21)

```
src/smtp_util.erl:71: Warning: crypto:rand_uniform/2 is deprecated and will be removed in a future release; use rand:uniform/1
src/smtp_util.erl:71: Warning: crypto:rand_uniform/2 is deprecated and will be removed in a future release; use rand:uniform/1
src/socket.erl:124: Warning: ssl:ssl_accept/1: deprecated; use ssl:handshake/1 instead
src/socket.erl:233: Warning: ssl:ssl_accept/3: deprecated; use ssl:handshake/3 instead
```

### PR DESCRIPTION
**Pull request by babariviere**
Update mix dependencies. Fixes most warning caused by old dependencies. (mostly `gen_smtp` who had a deprecation warning)

Closes #111

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -27,19 +27,19 @@ defmodule BambooSmtp.Mixfile do
...
-      {:bamboo, "~> 1.2"},
-      {:credo, "~> 1.4.0", only: [:dev, :test]},
+      {:bamboo, "~> 1.6"},
+      {:credo, "~> 1.4.1", only: [:dev, :test]},
       {:earmark, ">= 1.3.2", only: :docs},
-      {:excoveralls, "~> 0.12.0", only: :test},
+      {:excoveralls, "~> 0.13.3", only: :test},
       {:ex_doc, ex_doc_version(), only: :docs},
-      {:gen_smtp, "~> 0.15.0"},
+      {:gen_smtp, "~> 1.0.1"},
```
Comment (julien-leclercq) [inline mix.exs:35]: shouldn't we bump `bamboo_smtp` as well if we do this ?

Comment (babariviere) [reply mix.exs:35]: Yeah sure, should it be a minor version upgrade?

Comment (julien-leclercq) [reply mix.exs:35]: actually not ! As long as we don't publish it's ok, but this is another PR I think

File: mix.exs [hunk 2]
```diff
@@ -27,19 +27,19 @@ defmodule BambooSmtp.Mixfile do
 
   defp deps do
     [
-      {:bamboo, "~> 1.2"},
-      {:credo, "~> 1.4.0", only: [:dev, :test]},
+      {:bamboo, "~> 1.6"},
```
Comment (arnaudmorisset) [inline mix.exs:30]: I think this one was updated in another PR so you may have a conflict here. 🙂

**Review by arnaudmorisset — APPROVED** [review]

**Pull request merged**